src/plugins/comments/comments.js

```diff
@@ -395,15 +400,28 @@ class Comments extends BasePlugin {
     if (!force && (!this.range.from || !this.editor.isVisible())) {
       return;
     }
+
+    // targetingPreviousRow is used when the requested row is hidden, and the editor needs to be positioned on the previous row's coords
+    let targetingPreviousRow = false;
+
     const { rootWindow } = this.hot;
     const { wtTable, wtOverlays, wtViewport } = this.hot.view.wt;
     const scrollableElement = wtOverlays.scrollableElement;
-    const TD = wtTable.getCell(this.range.from);
-    const row = this.range.from.row;
-    const column = this.range.from.col;
+    const visualRow = this.range.from.row;
+    const visualColumn = this.range.from.col;
+    const renderableRow = (this.hot.rowIndexMapper.getRenderableFromVisualIndex(visualRow) ??
+      ((targetingPreviousRow = true) && this.hot.rowIndexMapper.getRenderableFromVisualIndex(this.hot.rowIndexMapper.getFirstNotHiddenIndex(visualRow, -1))));
```

Comment:
Assigning the `targetingPreviousRow` variable within the expression is not a best idea for code readability. It took me some time to understand what is for. I'd rewrite this to something like this:
```js
  refreshEditor(force = false) {
    if (!force && (!this.range.from || !this.editor.isVisible())) {
      return;
    }

    const { rowIndexMapper, columnIndexMapper } = this.hot;
    const { row: visualRow, col: visualColumn } = this.range.from;

    let renderableRow = rowIndexMapper.getRenderableFromVisualIndex(visualRow);
    let renderableColumn = columnIndexMapper.getRenderableFromVisualIndex(visualColumn);
    // Used when the requested row is hidden, and the editor needs to be positioned on the previous row's coords.
    const targetingPreviousRow = renderableRow === null;

    if (renderableRow === null) {
      renderableRow = rowIndexMapper
        .getRenderableFromVisualIndex(rowIndexMapper.getFirstNotHiddenIndex(visualRow, -1));
    }

    if (renderableColumn === null) {
      renderableColumn = columnIndexMapper
        .getRenderableFromVisualIndex(columnIndexMapper.getFirstNotHiddenIndex(visualColumn, -1));
    }

    const isBeforeRenderedRows = renderableRow === null;
    const isBeforeRenderedColumns = renderableColumn === null;

    renderableRow = renderableRow ?? 0;
    renderableColumn = renderableColumn ?? 0;

    const { rootWindow, view: { wt } } = this.hot;
    const { wtTable, wtOverlays, wtViewport } = wt;
    const scrollableElement = wtOverlays.scrollableElement;

    const TD = wtTable.getCell({
      row: renderableRow,
      col: renderableColumn,
    });

    const cellOffset = offset(TD);
    const lastColWidth = isBeforeRenderedColumns ? 0 : wtTable.getStretchedColumnWidth(renderableColumn);
    const lastRowHeight = targetingPreviousRow && !isBeforeRenderedRows ? outerHeight(TD) : 0;
    let cellTopOffset = cellOffset.top;
    let cellLeftOffset = cellOffset.left;

    if (wtViewport.hasVerticalScroll() && scrollableElement !== rootWindow) {
      cellTopOffset -= wtOverlays.topOverlay.getScrollPosition();
    }

    if (wtViewport.hasHorizontalScroll() && scrollableElement !== rootWindow) {
      cellLeftOffset -= wtOverlays.leftOverlay.getScrollPosition();
    }

    const x = cellLeftOffset + lastColWidth;
    const y = cellTopOffset + lastRowHeight;

    const commentStyle = this.getCommentMeta(visualRow, visualColumn, META_STYLE);
    const readOnly = this.getCommentMeta(visualRow, visualColumn, META_READONLY);

    if (commentStyle) {
      this.editor.setSize(commentStyle.width, commentStyle.height);
    } else {
      this.editor.resetSize();
    }

    this.editor.setReadOnlyState(readOnly);
    this.editor.setPosition(x, y);
  }
```

I've tried to simplify the whole logic in the method body to be more understandable for future visitors 😄  (after applying the proposed changes all tests pass)

Reply:
Yeah, I tried to avoid the "declaration - logic - declaration - logic (...)" look, but I guess it's still more readable than my approach.